RoboJackets/igvc-software
# Add a hyper parameter tuning and tensorboard feature to the nn trai…

# Description
This PR contains several changes to the existing nn training code.

This PR does the following:
- Added a hyper-parameter tuning code
- Refactored the training code to enable the hyper-parameter tuning code
- Added a tensor board feature to visualize training process
- Added print statements to see the training timer and overall training loss

This PR Fixes the following issues #565 #549

# Testing steps (If relevant)
## Test Case 1
1. Please download the training and validation dataset from the Robojackets cloud
[https://cloud.robojackets.org/apps/files/?dir=/IGVC/Software/dataset/IGVC&fileid=338649](url)
2. Open your google colab and load the notebook named "igvc-unet"
3. Modify your cfg file based on your file path
4. Run the following code `python train.py --epoch 100 --save_model --save_interval 10`
5. Check your training result by `tensorbaord --logdir="your runs file path"`

Expectation: This PR allows our current training model to visualize and compare trained models with different hyper-parameters. This PR also refactors the training code. 

# Self Checklist
